2/step1_1_huge_memory_usage.py

```diff
@@ -0,0 +1,30 @@
+# Definition for singly-linked list.
+# class ListNode:
+#     def __init__(self, val=0, next=None):
+#         self.val = val
+#         self.next = next
+class Solution:
+    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
+        def list_to_int(node):
+            output_number = 0
+            base_number = 0
+            while node is not None:
+                output_number += node.val * 10**base_number
+                base_number += 1
+                node = node.next
+            return output_number
+
+        def int_to_list(input_number):
+            output_list_number = []
+            while input_number >= 0:
```

Comment:
こちら無限ループになってしまっていますね。
以下のように修正しinput_number = 0の例外処理をすれば正しく動きました。

```suggestion
            while input_number > 0:
```

Reply:
nanaeさん、バグのご指摘ありがとうございます。例外条件を見落としており大変助かりました。
上記修正とint_to_list関数内の冒頭で下記の例外処理をすることで動作を確認できました。

  if input_number == 0:  
                 return ListNode(0)